x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/ReplaceStatsNestedExpressionWithEval.java

```diff
@@ -15,25 +15,36 @@
 import org.elasticsearch.xpack.esql.expression.function.aggregate.AggregateFunction;
 import org.elasticsearch.xpack.esql.expression.function.grouping.GroupingFunction;
 import org.elasticsearch.xpack.esql.optimizer.LogicalPlanOptimizer;
-import org.elasticsearch.xpack.esql.plan.logical.Aggregate;
 import org.elasticsearch.xpack.esql.plan.logical.Eval;
 import org.elasticsearch.xpack.esql.plan.logical.LogicalPlan;
+import org.elasticsearch.xpack.esql.plan.logical.Stats;
 
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
 /**
- * Replace nested expressions inside an aggregate with synthetic eval (which end up being projected away by the aggregate).
- * stats sum(a + 1) by x % 2
+ * Replace nested expressions inside a {@link Stats} with synthetic eval.
+ * {@code STATS SUM(a + 1) BY x % 2}
  * becomes
- * eval `a + 1` = a + 1, `x % 2` = x % 2 | stats sum(`a+1`_ref) by `x % 2`_ref
+ * {@code EVAL `a + 1` = a + 1, `x % 2` = x % 2 | STATS SUM(`a+1`_ref) BY `x % 2`_ref}
```

Comment:
Out of scope, but we could write more correct queries if we try to make an example.
```suggestion
 * {@code EVAL `a + 1` = a + 1, `x % 2` = x % 2 | STATS SUM(`a + 1`) BY `x % 2`}
```